src/handlers.ts

```diff
@@ -67,6 +76,134 @@ async function postToArticle(ctx: RequestContext<ContextData>, post: Post) {
+        globaldb: ctx.get('globaldb'),
+    });
+
+    const likeId = apCtx.getObjectUri(Like, {
+        id: createHash('sha256').update(objectToLike.id!.href).digest('hex'),
+    });
+
+    const undoId = apCtx.getObjectUri(Undo, {
+        id: createHash('sha256').update(likeId.href).digest('hex'),
+    });
+
+    const likeToUndoJson = await ctx.get('globaldb').get([likeId.href]);
+    if (!likeToUndoJson) {
+        return new Response(null, {
+            status: 409
+        });
+    }
+
+    const likeToUndo = await Like.fromJsonLd(likeToUndoJson);
+
+    const actor = await apCtx.getActor(ACTOR_DEFAULT_HANDLE); // TODO This should be the actor making the request
+
+    const undo = new Undo({
+        id: undoId,
+        actor: actor,
+        object: likeToUndo,
+        to: PUBLIC_COLLECTION,
+        cc: apCtx.getFollowersUri(ACTOR_DEFAULT_HANDLE),
+    });
+    const undoJson = await undo.toJsonLd();
+    await ctx.get('globaldb').set([undo.id!.href], undoJson);
+
+    await removeFromList(ctx.get('db'), ['liked'], likeId!.href);
+    await ctx.get('globaldb').delete([likeId!.href]);
+
+    let attributionActor: Actor | null = null;
+    if (objectToLike.attributionId) {
+        attributionActor = await lookupActor(apCtx, objectToLike.attributionId.href);
+    }
+    if (attributionActor) {
+        apCtx.sendActivity({ handle: ACTOR_DEFAULT_HANDLE }, attributionActor, undo, {
+            preferSharedInbox: true
+        });
+    }
+
+    apCtx.sendActivity({ handle: ACTOR_DEFAULT_HANDLE }, 'followers', undo, {
+        preferSharedInbox: true
+    });
+    return new Response(JSON.stringify(undoJson), {
+        headers: {
+            'Content-Type': 'application/activity+json',
+        },
+        status: 200,
+    });
+}
+
+export async function likeAction(
+    ctx: Context<{ Variables: HonoContextVariables }>,
+) {
+    const id = ctx.req.param('id');
+    const objectToLike = await lookupObject(id);
+    if (!objectToLike) {
+        return new Response(null, {
+            status: 404
+        });
+    }
+    const apCtx = fedify.createContext(ctx.req.raw as Request, {
+        db: ctx.get('db'),
+        globaldb: ctx.get('globaldb'),
+    });
+
+    const likeId = apCtx.getObjectUri(Like, {
+        id: createHash('sha256').update(objectToLike.id!.href).digest('hex'),
+    });
+
+    if (await ctx.get('globaldb').get([likeId.href])) {
+        return new Response(null, {
+            status: 409
+        });
+    }
+
+    const actor = await apCtx.getActor(ACTOR_DEFAULT_HANDLE); // TODO This should be the actor making the request
+
+    const like = new Like({
+        id: likeId,
+        actor: actor,
+        object: objectToLike,
+        to: PUBLIC_COLLECTION,
+        cc: apCtx.getFollowersUri(ACTOR_DEFAULT_HANDLE),
+    });
+    const likeJson = await like.toJsonLd();
+    await ctx.get('globaldb').set([like.id!.href], likeJson);
+
+    await addToList(ctx.get('db'), ['liked'], like.id!.href);
```

Comment:
🤔 Should this be the ID of the liked object rather than the like activity itself? Then the liked dispatcher returns the objects liked and not the like activities themselves?

Spec says:

> This is a list of every object from all of the actor's Like activities, added as a [side effect]

Reply:
I did initially think that, but then in the liked dispatcher, fedify would error because it expects a return type of `Like[]` 😕 

I think we need to get fedify updated (and fixed) to handle returning objects from the liked dispatcher - I'll sort that next week